website/server/models/group.js

```diff
@@ -648,6 +648,27 @@ schema.methods.sendChat = function sendChat (options = {}) {
   return newChatMessage;
 };
 
+schema.methods.handleQuestInvitation = async function handleQuestInvitation (user, accept) {
+  // Handle quest invitation atomically (update only current member when still undecided)
+  // to prevent multiple concurrent requests overriding updates
+  // see https://github.com/HabitRPG/habitica/issues/11398
+  const Group = this.constructor;
+  const result = await Group.update(
+    {
+      _id: this._id,
+      [`quest.members.${user._id}`]: { $exists: true, $eq: null },
```

Comment:
I think for this check you can use this query https://docs.mongodb.com/manual/tutorial/query-for-null-fields/#type-check

Reply:
Yeah, seems that should do the same thing. I can change it.